store/iavlstore.go

```diff
@@ -244,6 +245,32 @@ func (st *IavlStore) Query(req abci.RequestQuery) (res abci.ResponseQuery) {
 		} else {
 			_, res.Value = tree.GetVersioned(key, res.Height)
 		}
+	case "/ics23-key":
+		key := req.Data // Data holds the key bytes
+		res.Key = key
+		if !st.VersionExists(res.Height) {
+			res.Log = cmn.ErrorWrap(iavl.ErrVersionDoesNotExist, "").Error()
+			break
+		}
+		_, res.Value = tree.GetVersioned(key, res.Height)
+
+		if !req.Prove {
+			break
+		}
+
+		// Continue to prove existence/absence of value
+		// Must convert store.Tree to iavl.MutableTree with given version to use in CreateProof
+		iTree, err := tree.GetImmutable(res.Height)
+		if err != nil {
+			// sanity check: If value for given version was retrieved, immutable tree must also be retrievable
+			panic(fmt.Sprintf("version exists in store but could not retrieve corresponding versioned tree in store, %s", err.Error()))
+		}
+		mtree := &iavl.MutableTree{
+			ImmutableTree: iTree,
+		}
+
+		// get proof from tree and convert to merkle.Proof before adding to result
+		res.Proof = getProofFromTree(mtree, req.Data, res.Value != nil)
```

Comment:
Where do we return the store commit version for composing proof in BSC side?

Reply:
the version is the height itself, which is passed by the querier

Reply:
I doubt it.
```
type CommitInfo struct {
	Version int64
	StoreInfos []StoreInfo
}

type StoreInfo struct {
	Name string
	Core StoreCore
}

type StoreCore struct {
	CommitID CommitID
}

type CommitID struct {
	Version int64
	Hash    []byte
}
``` 

The version in CommitInfo is the height, but what we need is the one in CommitID.
I didn't check the details yet, but I guess it should be the latest change height of the store.
It's better to double-check that.